lib/primo/search.rb

```diff
@@ -51,7 +51,22 @@ def self.get(params = {})
       params = params.to_h.transform_keys(&:to_sym)
       method = get_method params
       (url, query) = url(params)
-      new super(url, query:, timeout: Primo.configuration.timeout(params)), method
+
+      unless Primo.configuration.enable_retries
+        new super(url, query:, timeout: Primo.configuration.timeout(params)), method
+      else
+        retry_count = Primo.configuration.retries
+        begin
+          new super(url, query:, timeout: Primo.configuration.timeout(params)), method
+        rescue Net::ReadTimeout
+          Primo.configuration.logger.warn("Primo request timed out. Retries remaining #{retry_count}")
+          if (retry_count -= 1) > 0
```

Comment:
When checking wether or not to do a retry you can/should also check if retries are configured to happen
```suggestion
          retries_enabled = Primo.configuration.enable_retries
          retries_left =  (retry_count -= 1) > 0
          if retries_enabled && retries_left
```